src/library/Uno.Material/Styles/Controls/v2/TextBox.xaml

```diff
@@ -396,22 +453,25 @@
 								<ColumnDefinition Width="Auto" />
 								<ColumnDefinition Width="*" />
 								<ColumnDefinition Width="Auto" />
+								<ColumnDefinition Width="Auto" />
 							</Grid.ColumnDefinitions>
 
 							<!-- Border in place to properly vertically center the icon inside when it's a one-line TextBox -->
 							<!-- but keep it in the same place and at the top when it's a multiline TextBox -->
 							<Border Height="26"
+									Width="40"
 									VerticalAlignment="Top"
-									Visibility="{Binding Path=(um:ControlExtensions.Icon), RelativeSource={RelativeSource TemplatedParent}, Converter={StaticResource MaterialNullToCollapsedConverter}, FallbackValue=Collapsed, TargetNullValue=Collapsed}">
-								<ContentPresenter x:Name="IconPresenter"
-												  Content="{Binding Path=(um:ControlExtensions.Icon), RelativeSource={RelativeSource TemplatedParent}}"
-												  HorizontalAlignment="Center"
-												  Margin="0,0,8,0"
-												  MaxHeight="34"
-												  MaxWidth="34"
-												  MinWidth="25"
-												  Foreground="{ThemeResource TextBoxLeadingIconForeground}"
-												  VerticalAlignment="{TemplateBinding VerticalContentAlignment}" />
+									Visibility="{Binding Path=(um:ControlExtensions.LeadingIcon), RelativeSource={RelativeSource TemplatedParent}, Converter={StaticResource MaterialNullToCollapsedConverter}, FallbackValue=Collapsed, TargetNullValue=Collapsed}">
+
+								<Button x:Name="LeadingIconButton"
+										Content="{Binding Path=(um:ControlExtensions.LeadingIcon), RelativeSource={RelativeSource TemplatedParent}}"
+										Command="{Binding Path=(um:ControlExtensions.LeadingCommand), RelativeSource={RelativeSource TemplatedParent}}"
+										IsHitTestVisible="{Binding Path=(um:ControlExtensions.LeadingCommand), RelativeSource={RelativeSource TemplatedParent}, Converter={StaticResource MaterialNullToFalseConverter}, FallbackValue=False, TargetNullValue=False}"
```

Comment:
Why do we need this for IsHitTestVisible? Is there a crash or something if you click on the button when no command is set?

Reply:
No, there is no crash. It's just to make sure it's not clickable/keyboard focusable when there is no IconCommand set

Reply:
Just re-tested it and it's actually not working as I was able to focus the button/icon pressing Tab, even if I set `IsHitTestVisible = False`

Reply:
Yeah I don't think it's necessary, the button can still be clickable/focusable. It'll just do nothing

Reply:
Done ;D